### ISSUE TITLE
Broken links in the doc's section

### ISSUE THREAD

**Issue by ayumi-cloud** [octobercms/docs#427]
Will slowly go through and explain each one.

URL: https://octobercms.com/docs/api/backend/widgets/table

![image](https://user-images.githubusercontent.com/57409060/71560641-ce08c980-2a64-11ea-9a97-77d38cb1ec8b.png)

Change to URL: https://jsperf.com/performance-of-array-vs-object/3

---

URL: https://octobercms.com/docs/api/october/rain/database/readme

![image](https://user-images.githubusercontent.com/57409060/71560671-fabce100-2a64-11ea-84ff-0853ddc00349.png)

No such file in this repo! Admins to look deeper please.

---

URL: https://octobercms.com/docs/api/october/rain/support/readme

![image](https://user-images.githubusercontent.com/57409060/71560705-5b4c1e00-2a65-11ea-9b2e-f9924f5c62b3.png)

No such file in this repo! Admins to look deeper please.

---

URL: https://octobercms.com/docs/cms/pages/

![image](https://user-images.githubusercontent.com/57409060/71560771-eaf1cc80-2a65-11ea-9182-17a25a530d48.png)

![image](https://user-images.githubusercontent.com/57409060/71560800-25f40000-2a66-11ea-84f1-ff79d5b635f9.png)

![image](https://user-images.githubusercontent.com/57409060/71560818-58056200-2a66-11ea-9c16-d669617439ce.png)

![image](https://user-images.githubusercontent.com/57409060/71560835-85521000-2a66-11ea-955d-5968ad447d77.png)

![image](https://user-images.githubusercontent.com/57409060/71560845-b3cfeb00-2a66-11ea-8bb9-a7327e05e07c.png)

Problem with the coding.

---

URL: https://octobercms.com/docs/help/using-composer

![image](https://user-images.githubusercontent.com/57409060/71560855-ebd72e00-2a66-11ea-832a-67cd1502ba50.png)

![image](https://user-images.githubusercontent.com/57409060/71560864-027d8500-2a67-11ea-969c-ec6513b5af5b.png)

Correct these two, please

---

URL: https://octobercms.com/docs/services/queues

![image](https://user-images.githubusercontent.com/57409060/71560878-2d67d900-2a67-11ea-904b-7bdc2fd2068c.png)

---

p.s. Links in orange are 404.

Thanks.

**Comment by daftspunk** [comment]
@ayumi-cloud Have you confirmed these reports? Upon inspection, most of these links appear to be working fine

**Comment by ayumi-cloud** [comment]
@daftspunk All the ones I have posted, I have personally checked and confirmed.

I'm guessing the https://octobercms.com/docs/cms/pages/ has the same issue, I found in this issue: https://github.com/octobercms/docs/issues/392

I can demo the error in a third testing tool, see gif:

![ezgif-5-ff37afb66fb1](https://user-images.githubusercontent.com/57409060/71564065-1c808d00-2a92-11ea-8f5c-b480802ec87b.gif)

clearer screenshot:

![image](https://user-images.githubusercontent.com/57409060/71564074-42a62d00-2a92-11ea-968c-b71f98c25ece.png)

**Comment by daftspunk** [comment]
I mean, have you opened the website in a browser and clicked the link yourself

https://octobercms.com/docs/cms/pages/ redirects to https://octobercms.com/docs/cms/pages which does not have this problem

**Comment by ayumi-cloud** [comment]
@daftspunk yes, have opened all the links in a browser myself.

I give you examples so we are on same page.

Link | Result
---|---
(main) https://octobercms.com/docs/cms/pages/ | 200
(ToC) https://octobercms.com/docs/cms/pages/#introduction | 200
(Internal) https://octobercms.com/docs/cms/pages/themes#php-section | 400

![image](https://user-images.githubusercontent.com/57409060/71585815-c7835c00-2b0f-11ea-9b3b-0dc44d4a62ee.png)

another example

![image](https://user-images.githubusercontent.com/57409060/71585939-38c30f00-2b10-11ea-9d2b-5f39e9edb86f.png)

etc.

**Comment by bennothommo** [comment]
@ayumi-cloud The links work for me just fine, but I think I see where the issue might lie.

The **PHP section** link you reported above as a 404 has the following HTML: `<a href="themes#php-section">PHP section</a>`. It's possible that your link scanner is appending the `themes` as a new segment at the end of the URL, as opposed to overwriting the last segment.

Instead of https://octobercms.com/docs/cms/pages/themes#php-section, it should be https://octobercms.com/docs/cms/themes#php-section

**Comment by ayumi-cloud** [comment]
@bennothommo If it was purely the link scanner as the issue that would be cool, the trouble is that I'm copying the links and if I click on those links in my browser I get the same error.

Yes the error is:

> Instead of https://octobercms.com/docs/cms/pages/themes#php-section, it should be https://octobercms.com/docs/cms/themes#php-section

Surely that's a rewrite issue in the redirects somewhere?

----

Also don't forget there are other types of 404 errors I have listed above e.g.

URL: https://octobercms.com/docs/api/october/rain/support/readme

![image](https://user-images.githubusercontent.com/57409060/71670719-b5e8c100-2d68-11ea-9e17-6b918e7fe443.png)

Gives me a link of: https://github.com/octobercms/docs/blob/master/advanced-console.md

Which results in a 404.

----

I have just noticed there is another coding issue on that web page as well.

![image](https://user-images.githubusercontent.com/57409060/71670789-ff391080-2d68-11ea-89a0-9624d7c2dbf8.png)

Notice the header sizes getting smaller and smaller due to an open `<code>` tag and not a matching closing `</code>` in the source code.

**Comment by bennothommo** [comment]
> @bennothommo If it was purely the link scanner as the issue that would be cool, the trouble is that I'm copying the links and if I click on those links in my browser I get the same error.
> 
> Yes the error is:
> 
> > Instead of https://octobercms.com/docs/cms/pages/themes#php-section, it should be https://octobercms.com/docs/cms/themes#php-section
> 
> Surely that's a rewrite issue in the redirects somewhere?

@ayumi-cloud I don't believe that's the case - what I meant in my original post was that I traversed the docs pages as you said, and clicked the links you pointed to, and they worked. Yes, the URL you listed above showed a 404, but unless I am clicking the wrong link somewhere, I never got that URL through normal navigation of the docs.

> 
> Also don't forget there are other types of 404 errors I have listed above e.g.
> 
> URL: https://octobercms.com/docs/api/october/rain/support/readme
> 
> ![image](https://user-images.githubusercontent.com/57409060/71670719-b5e8c100-2d68-11ea-9e17-6b918e7fe443.png)
> 
> Gives me a link of: https://github.com/octobercms/docs/blob/master/advanced-console.md
> 
> Which results in a 404.
> 

You're correct here though - the link there is definitely incorrect.

> URL: https://octobercms.com/docs/api/october/rain/support/readme
> 
> I have just noticed there is another coding issue on that web page as well.
> 
> ![image](https://user-images.githubusercontent.com/57409060/71670789-ff391080-2d68-11ea-89a0-9624d7c2dbf8.png)
> 
> another screenshot of same web page:
> 
> ![image](https://user-images.githubusercontent.com/57409060/71670992-a5851600-2d69-11ea-8354-ec32812e6192.png)
> 
> Notice the header sizes getting smaller and smaller due to an open `<code>` tag wrapping the header and text etc.

Would you mind moving this reported issue into its own issue on GitHub just so we don't conflate different issues in the same report?

**Comment by ayumi-cloud** [comment]
@bennothommo 

> Would you mind moving this reported issue into its own issue on GitHub just so we don't conflate different issues in the same report?

No problem, will do that now for you.
